### PR TITLE
Change GetPageSize to use Environment.SystemPageSize

### LINKED ISSUE #4258: `MemoryBlock.GetPageSize` should detect the correct page size for the platform
This function was added mainly for Apple Silicon which has 16 KB pages. However right now it just assumes that if it's macOS and Arm64, the page size is 16 KB, otherwise it is 4 KB. We should instead call the appropriate functions per platform and get the correct page size.

This will be required to make the emulator work on Asahi Linux which AFAIK also has 16 KB pages.

### PR DESCRIPTION
**Pull request by gnisman**
Change GetPageSize to use Environment.SystemPageSize #4258

Tested Environment.SystemPageSize on M1 mac and Windows 11 x64, the results are 16384 and 4096
Not sure how it will behave on SteamOS 
Closes #4258

### CODE REVIEW

File: Ryujinx.Memory/MemoryBlock.cs [hunk 1]
```diff
@@ -435,12 +435,7 @@ public static bool SupportsFlags(MemoryAllocationFlags flags)
 
         public static ulong GetPageSize()
         {
-            if (OperatingSystem.IsMacOS() && RuntimeInformation.ProcessArchitecture == Architecture.Arm64)
-            {
-                return 1UL << 14;
-            }
-
-            return 1UL << 12;
+            return (ulong) Environment.SystemPageSize;
```
Comment (gdkchan) [inline Ryujinx.Memory/MemoryBlock.cs:438]:
```suggestion
            return (ulong)Environment.SystemPageSize;
```

**Review by gdkchan — COMMENTED** [review]
Didn't know about `Environment.SystemPageSize`.

**Comment by gdkchan** [comment]
Can you include `Closes #4258` to make it close automatically on merge?

**Review by riperiperi — APPROVED** [review]
Nice, I also didn't know about this.

**Review by gdkchan — APPROVED** [review]
lgtm, thanks.

**Pull request merged**